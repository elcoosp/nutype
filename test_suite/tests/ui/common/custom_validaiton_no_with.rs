use nutype::nutype;

#[nutype(
    validate(error = NumError)
)]
pub struct Num(i32);

#[derive(Debug)]
enum NumError {
    TooBig,
    TooSmall,
}

fn main () {}
