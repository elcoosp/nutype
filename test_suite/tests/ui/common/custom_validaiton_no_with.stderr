error: The `error` attribute requires an accompanying `with` attribute.
       Please provide the validation function that returns Result<(), NumError>.
 --> tests/ui/common/custom_validaiton_no_with.rs:4:30
  |
4 |     validate(error = NumError)
  |                              ^
